### PR TITLE
Static Java parser, panic mode, input skipped, and separate error locations

### PR DESCRIPTION
**Pull request by ltratt**
This PR does four things.

First, it uses a static version of the Java parser (https://github.com/softdevteam/error_recovery_experiment/commit/1cffa45732c6eb08ec51d07c910349a8ca0fabfe) which should make running the experiment significantly faster. Roughly speaking we spent about 0.1s per file processing the grammar; given that the mean time for recovery per file is about 0.01s, this should reduce a really big bottleneck. As a double bonus, our static version doesn't bother printing out the parse tree which, for big files, was also a big slowdown.

Second, integrate the new panic mode recoverer (https://github.com/softdevteam/error_recovery_experiment/commit/8ff9295b4ddf01f5f6df48b5770145012a86597d), as a baseline to compare CPCT+ and MF against.

Third, count how much of the overall input is not parsed (either because of Delete repairs, or because the recoverer wasn't able to find a repair, leaving the rest of the file unparsed) (https://github.com/softdevteam/error_recovery_experiment/commit/2332e2ac9ec769511d6471ce2df8f1d41a9e3278). The hypothesis here is that the more input is skipped, the more cascading errors occur. I expected this relationship to be roughly linear; the early data (see a push to `error_recovery_paper`) suggests it might be non-linear. It'll be interesting to see what happens with a larger quantity of data.

Fourth, separate out the notion of the number of costs, and the number of error locations. The panic recoverer has no sense of costs, which highlights the difference between them. In retrospect, I should never have conflated these two things. So count error locations properly (https://github.com/softdevteam/error_recovery_experiment/commit/aeb7c8f02afcddd38599b907bbeededed4a45167), which allows us to give slightly better stats even for non-panic recoverers.

### CODE REVIEW

File: runner/process.py [hunk 1]
```diff
@@ -75,6 +77,7 @@ def __init__(self,
         sys.stdout.write(" input skipped...")
         sys.stdout.flush()
         self.input_skipped_ci = confidence_slice(self.bootstrap_input_skipped(), "0.99")
+        print
```
Comment (ptersilie) [inline runner/process.py]: Forgotten print?

Comment (ltratt) [reply runner/process.py]: It was accidentally removed at some point: without it, it's really difficult to work out when the processing script has moved to processing the next input file. Given that this takes many minutes (for the full dataset, hours) per recoverer, it's nice to know how it's going :)

Comment (ptersilie) [reply runner/process.py]: Sounds fair enough.

**Comment by ptersilie** [comment]
Looks good. Anything you want to squash?

**Comment by ltratt** [comment]
If you're OK with it as it is, I think it's OK as-is.

**Comment by ptersilie** [comment]
I thought maybe https://github.com/softdevteam/error_recovery_experiment/pull/14/commits/e1f35afafc1a1014e8b64b8b625f4007976940a7 but am also happy as is.

**Comment by ltratt** [comment]
That's sort-of a quite separate commit from everything else (and, to my astonishment, it seems to be working! I've tried so many things and failed in this regard).

**Review by ptersilie — APPROVED** [review]

**Comment by ptersilie** [comment]
Fair. In that case: merged!

**Pull request merged**